### PR TITLE
fix: `GET: /config` return all configurations when running standalone

### LINKED ISSUE #2584: `/config` API only returns frontend config under standalone mode
### What type of bug is this?

Configuration

### What subsystems are affected?

Standalone mode

### What happened?

`/config` only returns frontend config, missing configurations like `storage`, `region_engine`...

### What operating system did you use?

Unrelated

### Relevant log output and stack trace

```bash
mode = "standalone"

[heartbeat]
interval_millis = 18000
retry_interval_millis = 3000

[http]
addr = "127.0.0.1:4000"
timeout = "30s"
body_limit = "64MiB"

[grpc]
addr = "127.0.0.1:4001"
runtime_size = 8
max_recv_message_size = 536870912
max_send_message_size = 536870912

[mysql]
enable = true
addr = "127.0.0.1:4002"
runtime_size = 2

[mysql.tls]
mode = "disable"
cert_path = ""
key_path = ""

[postgres]
enable = true
addr = "127.0.0.1:4003"
runtime_size = 2

[postgres.tls]
mode = "disable"
cert_path = ""
key_path = ""

[opentsdb]
enable = true
addr = "127.0.0.1:4242"
runtime_size = 2

[influxdb]
enable = true

[prom_store]
enable = true

[otlp]
enable = true

[logging]
dir = "/tmp/greptimedb/logs"
enable_jaeger_tracing = false

[datanode.client]
timeout = "10s"
connect_timeout = "1s"
tcp_nodelay = true
```

### How can we reproduce the bug?

```bash
$ ./greptime standalone start

$ curl 127.0.0.1:4000/config
```

### PR DESCRIPTION
**Pull request by NiwakaDev**
I hereby agree to the terms of the [GreptimeDB CLA](https://gist.github.com/xtang/6378857777706e568c1949c7578592cc)

## What's changed and what's your intention?

Fixes #2584.

This PR makes frontend http-server

* return frontend config when running distributed mode
* return standalone config when running standalone mode
## Checklist

- [ ]  I have written the necessary rustdoc comments.
- [ ]  I have added the necessary unit tests and integration tests.

## Refer to a related PR or issue link (optional)
close #2584

### CODE REVIEW

**Comment by waynexia** [comment]
cc @etolbakov, you might be interested in this

**Comment by killme2008** [comment]
@NiwakaDev Sorry for being so late to review this PR. There are some conflicts that have to be resolved.

**Review by killme2008 — APPROVED** [review]
LGTM

File: src/cmd/src/options.rs [hunk 1]
```diff
@@ -36,6 +37,18 @@ pub struct MixOptions {
...
 
+impl From<MixOptions> for FrontendOptions {
+    fn from(value: MixOptions) -> Self {
+        value.frontend
+    }
+}
+
+impl TomlSerializable for MixOptions {
+    fn to_toml(&self) -> String {
+        toml::to_string(self).unwrap()
```
Comment (evenyag) [inline src/cmd/src/options.rs:48]: Better to return an error or a default string with the error. As we don't want to panic here.
```rust
match toml::to_string() {
    Ok(s) => s,
    Err(e) => format!("Failed to serialize options to TOML: {e}"),
}
```

File: src/frontend/src/server.rs [hunk 2]
```diff
@@ -44,13 +44,15 @@ pub type ServerHandler = (Box<dyn Server>, SocketAddr);
 
 impl Services {
     pub(crate) async fn build<T>(
-        opts: &FrontendOptions,
+        opts: impl Into<FrontendOptions> + TomlSerializable,
         instance: Arc<T>,
         plugins: Plugins,
     ) -> Result<ServerHandlers>
     where
         T: FrontendInstance,
```
Comment (evenyag) [inline src/frontend/src/server.rs:52]: style: Better to use the same style
```rust
pub(crate) async fn build<T, O>(opts: O, instance: Arc<T>, plugins: Plugins) -> Result<ServerHandlers>
    where:
        T: FrontendInstance,
        O: Into<FrontendOptions> + TomlSerializable
{}
```

File: src/frontend/src/frontend.rs [hunk 3]
```diff
@@ -76,6 +76,16 @@ impl FrontendOptions {
     }
 }
 
+pub trait TomlSerializable {
+    fn to_toml(&self) -> String;
+}
+
+impl TomlSerializable for FrontendOptions {
+    fn to_toml(&self) -> String {
+        toml::to_string(&self).unwrap()
```
Comment (evenyag) [inline src/frontend/src/frontend.rs:85]: Avoid panic

**Review by waynexia — APPROVED** [review]
Others LGTM

**Comment by NiwakaDev** [comment]
@evenyag  PTAL.

I guess that I addressed your review.

**Pull request merged**